core/src/main/java/org/apache/calcite/sql/type/SqlTypeAssignmentRule.java

```diff
@@ -60,6 +60,9 @@ private SqlTypeAssignmentRule(
     for (SqlTypeName interval : SqlTypeName.YEAR_INTERVAL_TYPES) {
       rules.add(interval, SqlTypeName.YEAR_INTERVAL_TYPES);
     }
+    for (SqlTypeName interval : SqlTypeName.WEEK_INTERVAL_TYPES) {
+      rules.add(interval, SqlTypeName.WEEK_INTERVAL_TYPES);
+    }
```

Comment:
To confirm, this is just saying that you can assign all week intervals types to other week interval types?

Reply:
Would this cause issues with assigning WEEK to DAY, since they're both in the SqlTypeName.WEEK_INTERVAL_TYPES group? Is that intended behavior?

Reply:
I think so yes.

I'm going to remove DAY from the SqlTypeFamily then. I guess it is strange to just have a family of a single type but I believe a SqlTypeFamily is required.